src/melee.cpp

```diff
@@ -1406,6 +1410,13 @@ bool player::block_hit(Creature *source, body_part &bp_hit, damage_instance &dam
         melee_attack( *source, false, tec );
     }
 
+    // give stamina penalty for blocking, heavier weapons and arm encumbrance
+    // are more tiring, while higher block bonus means less effort expended
+    int encumbrance_cost = roll_remainder( ( encumb( bp_arm_l ) + encumb( bp_arm_r ) ) / 5.0f );
+    int weight_cost = shield.weight() / ( 20 * std::max( 1, str_cur ) );
+    int mod_sta_block = ( weight_cost + encumbrance_cost + 20 - blocking_ability( shield ) ) * -1;
+    mod_stat( "stamina", std::min( -5, mod_sta_block ) );
+
```

Comment:
> Some feed back for player would be helpful here, otherwise people will complain why their stamina is gone.

Feedback here is not necessarily needed. We have stamina drained when running, and there is no feedback. We have stamina drained when fighting with melee weapons, and there is no feedback. And so on. Sometimes it is obvious, and doesn't need an explicit commentary.